readthedocs/core/resolver.py

```diff
@@ -149,15 +149,22 @@ def resolve(self, project, protocol='http', filename='', private=None,
 
         # This duplication from resolve_domain is for performance purposes
         # in order to check whether a custom domain should be HTTPS
-        if custom_domain:
+        use_custom_domain = any([
+            (custom_domain and not require_https_domain),
+            (custom_domain and require_https_domain and custom_domain.https),
+        ])
+        if use_custom_domain:
             domain = custom_domain.domain
         elif self._use_subdomain():
             domain = self._get_project_subdomain(canonical_project)
         else:
             domain = getattr(settings, 'PRODUCTION_DOMAIN')
 
+        protocol = 'http'
```

Comment:
I don't believe this logic is quite right. If `require_https_domain=True`, the protocol should be `https`. It doesn't matter whether there's a custom domain or the public domain is used.

Reply:
Maybe this flag should just be `require_https`. I don't know what `_domain` signifies.

Reply:
You are right! I'm fixing this.